src/main/java/com/zendesk/maxwell/row/RowMapDeserializer.java

```diff
@@ -78,20 +88,60 @@ public RowMap deserialize(JsonParser jsonParser, DeserializationContext deserial
 					}
 				}
 			}
+		} else if (data.isTextual()){
+			String decryptedData = RowEncrypt.decrypt(data.textValue(), this.encryption_key, this.secret_key);
+			JsonNode decryptedDataNode = mapper.readTree(decryptedData);
+			if (decryptedDataNode instanceof ObjectNode) {
+				Iterator keys = decryptedDataNode.fieldNames();
+				if (keys != null) {
+					while (keys.hasNext()) {
+						String key = (String) keys.next();
+						JsonNode value = decryptedDataNode.get(key);
+						if (value.isValueNode()) {
+							ValueNode valueNode = (ValueNode) value;
+							rowMap.putData(key, getValue(valueNode));
+						}
+					}
+				}
+			}
+			else{
+				throw new ParseException("`data` cannot be parsed.");
+			}
 		} else {
-			throw new ParseException("`data` is required and cannot be null.");
+			throw new ParseException("`data` cannot be parsed.");
 		}
 
-		if (oldData instanceof ObjectNode) {
-			Iterator keys = oldData.fieldNames();
-			if (keys != null) {
-				while (keys.hasNext()) {
-					String key = (String) keys.next();
-					JsonNode value = oldData.get(key);
-					if (value.isValueNode()) {
-						ValueNode valueNode = (ValueNode) value;
-						rowMap.putOldData(key, getValue(valueNode));
+		if (oldData != null) {
+			if (oldData instanceof ObjectNode) {
+				Iterator keys = oldData.fieldNames();
+				if (keys != null) {
+					while (keys.hasNext()) {
+						String key = (String) keys.next();
+						JsonNode value = oldData.get(key);
+						if (value.isValueNode()) {
+							ValueNode valueNode = (ValueNode) value;
+							rowMap.putOldData(key, getValue(valueNode));
+						}
+					}
+				}
+			} else if (oldData.isTextual()) {
+				String decryptedData = RowEncrypt
+						.decrypt(oldData.textValue(), this.encryption_key, this.secret_key);
+				JsonNode decryptedDataNode = mapper.readTree(decryptedData);
+				if (decryptedDataNode instanceof ObjectNode) {
+					Iterator keys = decryptedDataNode.fieldNames();
+					if (keys != null) {
+						while (keys.hasNext()) {
+							String key = (String) keys.next();
+							JsonNode value = decryptedDataNode.get(key);
+							if (value.isValueNode()) {
+								ValueNode valueNode = (ValueNode) value;
+								rowMap.putOldData(key, getValue(valueNode));
+							}
+						}
```

Comment:
This code seems a duplication from code above, can we make a method and reuse it?